client/post-editor/editor-term-selector/index.jsx

```diff
@@ -29,6 +31,14 @@ class EditorTermSelector extends Component {
 		this.boundOnTermsChange = this.onTermsChange.bind( this );
 	}
 
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.addedTerm && ( nextProps.addedTerm !== this.props.addedTerm ) ) {
+			const { siteId, postId, taxonomyName } = this.props;
```

Comment:
I think since you're dispatching based on incoming prop values, you'll want to destructure on `nextProps` (and call `nextProps.resetEditorTermAdded`).